src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -175,32 +170,39 @@ public static void Main()
             var compVerifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
             compVerifier.VerifyIL("C.Main",
 @"{
-  // Code size       36 (0x24)
+  // Code size       44 (0x2c)
   .maxstack  1
   .locals init (int V_0, //index
                 bool V_1,
-                int? V_2)
+                int? V_2,
+                int V_3)
   IL_0000:  nop
   IL_0001:  call       ""int? C.TryGet()""
   IL_0006:  stloc.2
   IL_0007:  ldloca.s   V_2
-  IL_0009:  call       ""int int?.GetValueOrDefault()""
-  IL_000e:  stloc.0
-  IL_000f:  ldloca.s   V_2
-  IL_0011:  call       ""bool int?.HasValue.get""
-  IL_0016:  stloc.1
-  IL_0017:  ldloc.1
-  IL_0018:  brfalse.s  IL_0023
-  IL_001a:  nop
-  IL_001b:  ldloc.0
-  IL_001c:  call       ""void System.Console.WriteLine(int)""
-  IL_0021:  nop
+  IL_0009:  call       ""bool int?.HasValue.get""
+  IL_000e:  brfalse.s  IL_001d
+  IL_0010:  ldloca.s   V_2
+  IL_0012:  call       ""int int?.GetValueOrDefault()""
+  IL_0017:  stloc.3
+  IL_0018:  ldloc.3
+  IL_0019:  stloc.0
+  IL_001a:  ldc.i4.1
```

Comment:
>ldc [](start = 12, length = 3)

Emit is unnecessarily realizing the boolean result into a temp. I'm working on fixing that now for the next PR.

Reply:
Ah, I see that is a result of the injected instrumentation code.

---
In reply to: [172710193](https://github.com/dotnet/roslyn/pull/24059#discussion_r172710193) [](ancestors = 172710193)